vendor/github.com/mindprince/gonvml/bindings.go

```diff
@@ -442,6 +442,35 @@ func (d Device) UtilizationRates() (uint, uint, error) {
 	return uint(utilization.gpu), uint(utilization.memory), errorString(r)
 }
 
+func (d Device) nvmlDeviceGetThroughput() (float64, error) {
+	if C.nvmlHandle == nil {
+		return 0, errLibraryNotLoaded
+	}
+	var fieldValues C.nvmlFieldValue_t
+	fieldValues.fieldId = C.NVML_FI_DEV_NVLINK_THROUGHPUT_DATA_TX
+	fieldValues.scopeId = 0
+	r := C.nvmlDeviceGetFieldValues(d.dev, 1, &fieldValues)
+	if r.nvmlReturn != C.NVML_SUCCESS {
+		return 0, errorString(r)
+	}
+	if r.valueType == C.NVML_VALUE_TYPE_DOUBLE {
```

Comment:
maybe we can simply the if else via using a dict

Reply:
Not sure what you mean by using a dict here. Also note that this particular change has been updated - in the latest commit, we need to cast the C union into an unsafe pointer to read it.